src/phpDocumentor/ConfigurationFactory.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright 2010-2015 Mike van Riel<mike@phpdoc.org>
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor;
+
+use phpDocumentor\ConfigurationFactory\Strategy;
+
+/**
+ * The ConfigurationFactory converts the configuration xml from a Uri into an array.
+ */
+final class ConfigurationFactory
+{
+    /**
+     * The Uri that contains the path to the configuration file.
+     *
+     * @var Uri
+     */
+    private $uri;
+
+    /**
+     * The configuration xml.
+     *
+     * @var \SimpleXMLElement
+     */
+    private $xml;
+
+    /**
+     * All strategies that are used by the ConfigurationFactory.
+     *
+     * @var Strategy[]
+     */
+    private $strategies = [];
+
+    /**
+     * Initializes the ConfigurationFactory.
+     *
+     * @param Strategy[] $strategies
+     * @param Uri        $uri
+     * @param string     $schemaPath
+     */
+    public function __construct(array $strategies, Uri $uri, $schemaPath = '')
+    {
+        foreach ($strategies as $strategy) {
+            $this->registerStrategy($strategy);
+        }
+
+        $this->replaceLocation($uri);
+    }
+
+    /**
+     * Replaces the location of the configuration file if it differs from the existing one.
+     *
+     * @param Uri $uri
+     *
+     * @return void
+     */
+    public function replaceLocation(Uri $uri)
+    {
+        if ($this->uri !== $uri) {
```

Comment:
This check will only verify if the $uri is the exact same object but not whether the URLs themselves are the same (the identity of a Value Object is in its properties). 

So either

1. `!==` should be replaced with `!=`
2. the URI should be extracted from both Value Objects and compared, or 
3. a `equals()` method should be introduced in the URI class that will enable a URI to know how to compare itself to another URI.

The latter is the preferred form